javalib/src/main/scala/java/lang/process/UnixProcess.scala

```diff
@@ -19,18 +19,27 @@ private[process] abstract class UnixProcessHandle extends GenericProcessHandle {
 }
 
 private[process] object UnixProcess {
+  def apply(
+      stdin: FileDescriptor,
+      stdout: FileDescriptor,
+      stderr: FileDescriptor
+  )(handle: UnixProcessHandle): GenericProcess = new GenericProcess(handle) {
+    override protected def fdIn: FileDescriptor = stdin
+    override protected def fdOut: FileDescriptor = stdout
+    override protected def fdErr: FileDescriptor = stderr
+  }
+
   def apply(
       handle: UnixProcessHandle,
       infds: Ptr[CInt],
       outfds: Ptr[CInt],
       errfds: Ptr[CInt]
-  ): GenericProcess = new GenericProcess(handle) {
-    override protected def fdIn = new FileDescriptor(!(infds + 1))
-    override protected def fdOut = new FileDescriptor(!outfds, readOnly = true)
-    override protected def fdErr =
-      if (null == errfds) new FileDescriptor()
-      else new FileDescriptor(!errfds, readOnly = true)
-  }
+  ): GenericProcess = apply(
```

Comment:
Author's prerogative but I wonder why the
first do `apply()` do not have the same form.
That is the second has the handle as the first argument
inside the first/only parentheses and the first has
the handle inside of the second parenthesis.

The first form gives better separation of argument
concerns.

The second constructor is probably the historical form and
the reasoning may be to keep from having to change `Windows` process.  

If change is in for a penny, may as well be consistent and be in for a pound.

Reply:
ah, once again, i feel that my instructions are confusing. you were not supposed to review the first commit in this PR. that same commit has its own PR, but I guess I am the only one who reviews PRs a commit at a time, rather than the entire mess altogether :)